Sources/Common/Communication/EventStorage.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+
+/// Defines the protocol for event storage management.
+///
+/// This protocol specifies the methods for storing, loading, and removing events,
+/// facilitating the management of event persistence.
+public protocol EventStorage: AutoMockable {
+    /// Stores an event asynchronously.
+    /// - Parameter event: The event to be stored.
+    func store(event: AnyEventRepresentable) async throws
+    /// Loads all events of a specific type asynchronously.
+    /// - Parameter type: The type of events to load.
+    /// - Returns: An array of events of the specified type.
+    func loadEvents(ofType type: String) async throws -> [AnyEventRepresentable]
+    /// Removes a specific event asynchronously by its storage identifier.
+    /// - Parameters:
+    ///   - eventType: The type of the event.
+    ///   - storageId: The unique identifier of the event to remove.
+    func remove(ofType eventType: String, withStorageId storageId: String) async
+}
+
+/// Errors related to event bus operations.
+enum EventBusError: Error {
+    case invalidEventType
+    case decodingError
+}
+
+/// An actor that manages event storage, providing thread-safe operations for storing, loading, and removing events.
+///
+/// This class handles the persistence of events using the file system, ensuring data integrity and consistency.
+// sourcery: InjectRegisterShared = "EventStorage"
+actor EventStorageManager: EventStorage {
+    private let fileManager = FileManager.default
+    public var baseDirectory: URL
+    private let logger: Logger
+    private let jsonAdapter: JsonAdapter
+
+    /// Initializes the EventStorageManager with the necessary dependencies.
+    /// - Parameters:
+    ///   - logger: A logger for logging information and errors.
+    ///   - jsonAdapter: A JSON adapter for encoding and decoding events.
+    init(logger: Logger, jsonAdapter: JsonAdapter) {
+        let documentDirectory = fileManager.urls(for: .documentDirectory, in: .userDomainMask)[0]
+        self.baseDirectory = documentDirectory.appendingPathComponent("Events")
+        self.logger = logger
+        self.jsonAdapter = jsonAdapter
+    }
+
+    /// Updates the base directory for event storage, primarily used for testing.
+    /// - Parameter baseDirectory: The new base directory for storing events.
+    func updateBaseDirectory(baseDirectory: URL) {
+        self.baseDirectory = baseDirectory
+    }
+
+    func store(event: AnyEventRepresentable) async throws {
```

Comment:
nit
```suggestion
    func store(event: AnyEventRepresentable) throws {
```

Reply:
since this method is performing file operation, i think it should be async